### ISSUE TITLE
Rename http related functions

### ISSUE THREAD

**Issue by smimram** [savonet/liquidsoap#1499]
So that we don't forget, Romain recently made the following comment:

I think we should start thinking about renaming all these:
- `input.http` is actually designed to pull icecast streams
- `input.harbor` really doesn't say anything about what's going on.
I'd suggest to consider a webserver name space for all harbor stuff and rename `input.http` as `input.icecast`.

I fully agree, in particular our users really don't need to remember that the internal name for our webserver is harbor...

**Comment by rrolla** [comment]
I agree about naming the same for abbreviations for example for list.hd much better will be list.head.

**Comment by toots** [comment]
@rrolla I agree as well. If you're interested, this is an area where we could definitely use a little PR help!

You can for instance change the name of `list.hd` here: https://github.com/savonet/liquidsoap/blob/main/libs/list.liq#L18

Then add a deprecation at the bottom of the file here: https://github.com/savonet/liquidsoap/blob/main/libs/deprecations.liq

Then the doc would need to be updated, which you could do with a `git sed` inside `doc/content`. Eventually, it'd be good to add a migration note here: https://github.com/savonet/liquidsoap/tree/main/doc/content

Then you can repeat that for each operator that you want to propose for renaming.

**Comment by rrolla** [comment]
> @rrolla I agree as well. If you're interested, this is an area where we could definitely use a little PR help!
> 
> You can for instance change the name of `list.hd` here: https://github.com/savonet/liquidsoap/blob/main/libs/list.liq#L18
> 
> Then add a deprecation at the bottom of the file here: https://github.com/savonet/liquidsoap/blob/main/libs/deprecations.liq
> 
> Then the doc would need to be updated, which you could do with a `git sed` inside `doc/content`. Eventually, it'd be good to add a migration note here: https://github.com/savonet/liquidsoap/tree/main/doc/content
> 
> Then you can repeat that for each operator that you want to propose for renaming.

Thanks, yeah maybe I can help with this from time to time.
